src/opm/parser/eclipse/EclipseState/Schedule/Action/ASTNode.cpp

```diff
@@ -106,16 +107,28 @@ Action::Value ASTNode::value(const Action::Context& context) const {
           The matching code is special case to handle one-argument cases with
           well patterns like 'P*'.
         */
-        if ((this->arg_list.size() == 1) && (arg_list[0].find("*") != std::string::npos)) {
+        if ((this->arg_list.size() == 1) && (this->arg_list[0].find("*") != std::string::npos)) {
             if (this->func_type != FuncType::well)
                 throw std::logic_error(": attempted to action-evaluate list not of type well.");
 
+            const auto& well_arg = this->arg_list[0];
             Action::Value well_values;
-            int fnmatch_flags = 0;
-            for (const auto& well : context.wells(this->func)) {
-                if (fnmatch(this->arg_list[0].c_str(), well.c_str(), fnmatch_flags) == 0)
-                    well_values.add_well(well, context.get(this->func, well));
+            std::vector<std::string> wnames;
+
+            if (well_arg[0] == '*' && well_arg.size() > 1) {
+                const auto& wlm = context.wlm();
+                const auto& wl = wlm.getList(well_arg);
```

Comment:
Could `well_arg` possibly be a well-list **template** here, e.g.,
```
well_arg = "*PROD*";
```
for all well lists named `"PROD<Something>"`?

Reply:
As I understand it you can have exactly one `'*' ` - if it is the first character it is the name of a well list, as implemented here, otherwise it is a wildcard pattern - all wells starting with "PROD*"  - that is already supported.

I might misunderstand what you mean?

Reply:
> I might misunderstand what you mean?

It's a bit obscure and not widely used, but it is definitely possible to have templates/wildcards for well list arguments, at least for the `W*` keywords in the `SCHEDULE` section (`WCONPROD`&c).  For instance, if we have the well lists `*PROD-E` and `*PROD-W`, then
```
*PROD-*
```
refers to both of them.

Reply:
This could in fact be even more elaborate with something like
```
**PROD*
```
that refers to all well lists that have the string `PROD` somewhere in their names.

Reply:
OK - I'll hack something up.